tests/test_xml.py

```diff
@@ -83,3 +83,57 @@ def test_invalid_xml(file):
     # No formatting done, data is simply printed as-is
     r = http(URL_EXAMPLE)
     assert xml_data in r
+
+
+@responses.activate
+def test_content_type_from_format_options_argument():
+    """Test server XML with a incorrect Content-Type header.
+    Using the --format-options to use the good one.
+    """
+    responses.add(responses.GET, URL_EXAMPLE, body=SAMPLE_XML_DATA,
+                  content_type='plain/text')
+
+    # Ensure the option is taken into account only for responses.
+    # Request
+    r = http('--offline', '--raw', SAMPLE_XML_DATA,
+             '--format-options', 'response.content_type:application/xml',
+             URL_EXAMPLE)
+    assert SAMPLE_XML_DATA in r
+
+    # Response
+    r = http('--format-options', 'response.content_type:application/xml',
+             URL_EXAMPLE)
+    assert pretty_xml(parse_xml(SAMPLE_XML_DATA)) in r
```

Comment:
Since this is used multiple times, let’s add a constant at the top, e.g.:

```python
XML_DATA_RAW = '<?xml version="1.0" encoding="utf-8"?><root><e>text</e></root>'
XML_DATA_FORMATTED = pretty_xml(parse_xml(SAMPLE_XML_DATA))
```